### PR TITLE
feat(workflow): suspend and resume cases and work items

### LINKED ISSUE #1156: RFC: Suspend / resume work items
To handle use cases where a work item (_or even a case_) needs to be suspended and resumed at another point we need two new mutations / api methods:

`suspendWorkItem` / `suspend_work_item`

- Takes a work item as parameter and sets the status of said work item to `SUSPENDED`
- Requires the work item to have the status `READY`
- Fires an event `suspended_work_item`

`resumeWorkItem` / `resume_work_item`

- Takes a work item as parameter and sets the status of said work item to `READY`
- Requires the work item to have the status `SUSPENDED`
- Fires an event `resumed_work_item`

Those new methods would follow an official pattern of the standard we try to follow: http://www.workflowpatterns.com/patterns/resource/detour/wrp32.php

### PR DESCRIPTION
**Pull request by anehx**
This adds functionality (Python API as well as GraphQL API) for
suspending and resuming a work item or even a whole case.

Closes #1156

### CODE REVIEW

**Comment by czosel** [comment]
Black complains that it would reformat `0024_auto_20200901_1026.py`.

**Review by czosel — APPROVED** [review]
:100:

File: caluma/caluma_workflow/serializers.py [hunk 1]
```diff
@@ -448,3 +510,63 @@ class Meta:
...
+
+    def validate(self, data):
+        try:
+            domain_logic.SuspendWorkItemLogic.validate_for_suspend(self.instance)
+        except ValidationError as e:
+            raise exceptions.ValidationError(str(e))
+
+        return super().validate(data)
+
+    def update(self, work_item, validated_data):
```
Comment (winged) [inline caluma/caluma_workflow/serializers.py:526]: Missing `@transaction.atomic`

File: caluma/caluma_workflow/serializers.py [hunk 2]
```diff
@@ -448,3 +510,63 @@ class Meta:
...
+
+    def validate(self, data):
+        try:
+            domain_logic.ResumeWorkItemLogic.validate_for_resume(self.instance)
+        except ValidationError as e:
+            raise exceptions.ValidationError(str(e))
+
+        return super().validate(data)
+
+    def update(self, work_item, validated_data):
```
Comment (winged) [inline caluma/caluma_workflow/serializers.py:556]: Missing `@transaction.atomic`

File: caluma/caluma_workflow/models.py [hunk 3]
```diff
@@ -167,13 +168,14 @@ class WorkItem(UUIDModel):
...
     STATUS_SKIPPED = "skipped"
+    STATUS_SUSPENDED = "suspended"
 
-    STATUS_CHOICES = (STATUS_READY, STATUS_COMPLETED, STATUS_CANCELED)
     STATUS_CHOICE_TUPLE = (
         (STATUS_READY, "Task is ready to be processed."),
         (STATUS_COMPLETED, "Task is done."),
         (STATUS_CANCELED, "Task is cancelled."),
         (STATUS_SKIPPED, "Task is skipped."),
+        (STATUS_SUSPENDED, "Task is suspended."),
```
Comment (winged) [inline caluma/caluma_workflow/models.py:178]: Just noticed due to this change's effect in the migrations: This is on the `WorkItem` model, not `Task`, thus should have work item in the description. Maybe add a drive-by fix commit?

**Review by winged — CHANGES_REQUESTED** [review]
Awesome work, I like it! :tada: 

Just some minor fixes required.

File: caluma/caluma_workflow/serializers.py (hunk 1 above)
Comment (anehx) [reply caluma/caluma_workflow/serializers.py:533]: True! I [fixed the missing transactions](c7963005fbfe1fabeded03ae8304a9cda6bc1239) in the whole file. Since we always trigger an event that could potentially do DB changes it should be used in every `update` or `create`.

File: caluma/caluma_workflow/serializers.py (hunk 2 above)
Comment (anehx) [reply caluma/caluma_workflow/serializers.py:564]: Fixed

File: caluma/caluma_workflow/models.py (hunk 3 above)
Comment (anehx) [reply caluma/caluma_workflow/models.py]: Absolutely, that bothered me as well. Fixed in bfed4981b73acfbe49bb1ab4644b5742d63058b3

**Review by winged — APPROVED** [review]
